### DISCUSSION TITLE
How to customize the Company Settings page?

### DISCUSSION THREAD

**Discussion by russdot** [andrewdwallo/filament-companies#156] [Q&A] [answered]
I'd like to customize the Company Settings page - specifically removing the "company owner". But I am having trouble determining exactly what I need to override? Thanks for any assistance.

**Comment by andrewdwallo** [comment, marked as answer]
You can create a different **"tenantProfile"** then the default one provided in the FilamentCompaniesServiceProvider.
https://filamentphp.com/docs/3.x/panels/tenancy#adding-a-tenant-profile-page

Here is an example of it being customized in my app: https://github.com/andrewdwallo/erpsaas/blob/3.x/app/Filament/Company/Pages/ManageCompany.php

**Reply by russdot** [reply]
Thanks! I used the method you suggested. I ran into an issue where the old form would display only after updating the company name (and would disappear on a page refresh)

When I changed the name of the view from `update-company-name-form` to `custom-update-company-name-form`, it resolved. Perhaps due to the component map in `FilamentCompaniesServiceProvider->configureComponents()`?

**Reply by russdot** [reply]
@andrewdwallo follow-up question regarding `FilamentCompaniesServiceProvider->configureComponents()`

I managed to override the livewire component map and want to confirm it is the correct way to do so (that won't interfere with the package)

1. Add `andrewdwallo/filament-companies` to `extra.laravel.dont-discover` in `composer.json` to prevent auto-discovery.
1. Create new `ServiceProvider` which extends `Wallo\FilamentCompanies\FilamentCompaniesServiceProvider`
1. Add this new `ServiceProvider` to `providers.php` (before the `PanelProvider`)
1. Override `ServiceProvider::configureComponents()` with new component for `update-company-name-form`

I now don't seem to encounter the issue where the old form displays after updating. Is this the correct approach? Thanks!

**Reply by andrewdwallo** [reply]
@russdot You shouldn't have to do this. Is your app public and available for me to look at?

**Reply by russdot** [reply]
Thanks for the quick response! I just tried to replicate on a fresh install and was unable to. If I can reproduce I'll toss it onto a public repo. If not, I may need to retrace some steps to ensure I'm not missing something. Thanks.

**Reply by russdot** [reply]
I suspect it may have been related to a cached view? (i.e. forgetting to run `artisan optimize:clear`)

**Reply by andrewdwallo** [reply]
> I suspect it may have been related to a cached view? (i.e. forgetting to run `artisan optimize:clear`)

Yeah that's possible.
